include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -105,11 +108,38 @@ namespace seqan3
  * \relates seqan3::aligned_sequence
  * \{
  */
+//!\}
+//!\cond
+template <typename t>
+SEQAN3_CONCEPT aligned_sequence =
+    sequence<t> &&
+    std::ranges::forward_range<t> &&
+    std::equality_comparable_with<std::ranges::range_reference_t<t>, gap const &>;
```

Comment:
> I would move the forward requirement into the writable concept.

Alright. When I wrote this I somehow thought it is needed.. but I can't remember why. I think you are right that it is only necessary in the writable case. 

Thanks for your detailed thoughts. It will help anyone who comes back to this at some point.